### ISSUE TITLE
Rename _typeshed.AnyPath?

### ISSUE THREAD

**Issue by srittau** [python/typeshed#5470]
TLDR: Let's rename `AnyPath` to `PathName` (or a better alias).

I seem to remember that @JelleZijlstra didn't like the name `AnyPath` as an alias for `str | bytes | PathLike[str] | PathLike[bytes]` when I introduced it. I agreed, but couldn't come up with a better name. When working on #5467, it became apparent to me that having a "real" `AnyPath`, i.e. an alias that is basically `AnyPath[_T] = _T | PathLike[_T]` would be quite useful. I don't think there is a way to have a generic alias yet, but I still propose we should rename `AnyPath` in preparation and to avoid confusion with `AnyStr`. The best I can come up with at the moment is `PathName`. Any better ideas? Other suggestions? Not a good idea?

**Comment by Akuli** [comment]
Tkinter stubs have a generic alias: https://github.com/python/typeshed/blob/4a3b6eb8d8a5f02ee08994635d324101899940ed/stdlib/tkinter/__init__.pyi#L100

**Comment by JelleZijlstra** [comment]
If only we could rename `AnyStr`. It would be better if typevars were suffixed with `T`.

> Tkinter stubs have a generic alias

I think mypy has issues with aliases of the form `T | Cls[T]`. Worth trying again though to see if that works now. In a couple of places we have added separate overloads for AnyStr and PathLike to work around that bug.

I'm not having much luck thinking of a better name that's still concise. `PathOfAnyKind`, `StrOrBytesPath`?

**Comment by srittau** [comment]
Maybe `StrOrBytesPath` is the best option. While not concise, it's clear, especially its relation to `StrPath` and `BytesPath`.

**Comment by srittau** [comment]
#5592 removes `AnyPath`. I'd suggest to give it a few months before introducing the `AnyPath` generic alias.

**Comment by AlexWaygood** [comment]
The original `AnyPath` has now been renamed, and we have a generic type alias in the form of `_typeshed.GenericPath`: #7970. Closing as completed 🎉
